### ISSUE TITLE
Lint that helps users realize that `&` is often the best way to create rectangles.

### ISSUE THREAD

**Issue by jacob314** [dart-lang/linter#2225]
This lint is for users of the Flutter framework to help them discover that the `&` operator often provides a cleaner way to define rectangles..

**Examples**
```dart
Rect.fromLTWH(offset.dx, offset.dy, size.width, size.height)
```
Should generate a lint hint that you should write
```dart
offset & size
```
instead.

**Additional context**
It is hard for users to discover that the `&` operator is often the right way to create a `Rect` instance. 
Autocomplete in Dart doesn't suggest operators and for autocomplete to ever help in this case, users would need to know to start with the offset. What we want is some way to suggest a lint or autocomplete style hint whenever a user creates a Rectangle that could be more easily created with `&` that they should.
<img width="357" alt="Screen Shot 2020-08-26 at 3 50 12 PM" src="https://user-images.githubusercontent.com/1226812/91373696-e2f05500-e7cb-11ea-8fb1-16bf75158b90.png">

**Comment by a14n** [comment]
You can also +1 dart-lang/sdk#32174 :)

**Comment by dnfield** [comment]
This might be unfortunate in the case where the Rect could have been const, but operators won't allow that.

For example:

```dart
const Rect rect = Rect.fromLTWH(offset.dx, offset.dy, size.width, size.height);
```

Is valid if `offset` and `size` are const, but 

```dart
const Rect rect = offset & size;
```

Is not valid.

**Comment by jacob314** [comment]
I can't reproduce that case.
```dart
const Offset offset = Offset(3.0, 4.0);
const Size size = Size(4.0, 5.0);
const Rect rect = Rect.fromLTWH(offset.dx, offset.dy, size.width, size.height);
```
Fails with the error:
"error: Arguments of a constant creation must be constant expressions. (const_with_non_constant_argument at...."

**Comment by dnfield** [comment]
Yeah, I edited it to fix - you have to use the `.dx` and `.dy` properties, since width and height are getters.

**Comment by jacob314** [comment]
Do you mean `._dx` and `.dy` for size?
`._dx` and `._dy` are private on the size object so I'm not clear how those can be used unless your code is in `package:flutter`

**Comment by dnfield** [comment]
Ahh ok - I guess this is impossible then. I don't know why I thought I had done it.
